mirgecom/euler.py

```diff
@@ -158,6 +182,165 @@ def euler_operator(dcoll, state, gas_model, boundaries, time=0.0,
+        # mixture pressure (used in the inviscid flux calculations) depends on
+        # temperature and we need to seed the temperature calculation for the
+        # (+) part of the partition boundary with the remote temperature data.
+        tseed_interior_pairs = [
+            # Get the interior trace pairs onto the surface quadrature
+            # discretization (if any)
+            interp_to_surf_quad(tpair)
+            for tpair in interior_trace_pairs(dcoll, state.temperature)
+        ]
+
+    def _interp_to_surf_modified_conservedvars(gamma, utpair):
+        # Takes a trace pair containing the projected entropy variables
+        # and converts them into conserved variables on the quadrature grid.
+        local_dd = utpair.dd
+        local_dd_quad = local_dd.with_discr_tag(quadrature_tag)
+        # Interpolate entropy variables to the surface quadrature grid
+        vtilde_tpair = op.project(dcoll, local_dd, local_dd_quad, utpair)
+        if isinstance(gamma, DOFArray):
+            gamma = op.project(dcoll, dd_vol, local_dd_quad, gamma)
+        return TracePair(
+            local_dd_quad,
+            # Convert interior and exterior states to conserved variables
+            interior=entropy_to_conservative_vars(gamma, vtilde_tpair.int),
+            exterior=entropy_to_conservative_vars(gamma, vtilde_tpair.ext)
+        )
+
+    cv_interior_pairs = [
+        # Compute interior trace pairs using modified conservative
+        # variables on the quadrature grid
+        # (obtaining state from projected entropy variables)
+        _interp_to_surf_modified_conservedvars(gamma, tpair)
+        for tpair in interior_trace_pairs(dcoll, entropy_vars)
+    ]
+
+    boundary_states = {
+        # TODO: Use modified conserved vars as the input state?
+        # Would need to make an "entropy-projection" variant
+        # of *project_fluid_state*
+        btag: project_fluid_state(
+            dcoll, dd_vol,
+            # Make sure we get the state on the quadrature grid
+            # restricted to the tag *btag*
+            as_dofdesc(btag).with_discr_tag(quadrature_tag),
+            state, gas_model) for btag in boundaries
+    }
+
+    # Interior interface state pairs consisting of modified conservative
+    # variables and the corresponding temperature seeds
+    interior_states = make_fluid_state_trace_pairs(cv_interior_pairs,
+                                                   gas_model,
+                                                   tseed_interior_pairs)
+
+    # Compute interface contributions
+    inviscid_flux_bnd = inviscid_flux_on_element_boundary(
+        dcoll, gas_model, boundaries, interior_states,
+        boundary_states, quadrature_tag=quadrature_tag,
+        numerical_flux_func=inviscid_numerical_flux_func, time=time,
+        dd=dd_vol)
+
+    return op.inverse_mass(
+        dcoll,
+        inviscid_vol_term - op.face_mass(dcoll, dd_allfaces_quad, inviscid_flux_bnd)
+    )
```

Comment:
Need `dd_vol` for `inverse_mass` and `face_mass` (or just use `div_operator` like `euler_operator` does.)

Reply:
Actually, we require this because `inviscid_vol_term` has already been computed with volume flux differencing, replacing the div op for that term.   Fixed up the dd specifications though!